### PR TITLE
implement an upper bound limit to the number of tracked executor

### PR DESCRIPTION
**Pull request by ImpSy**
## Purpose of this PR
In some case executor can enter in a crashlooping state causing the CR to grow out of control
This can have an impact on the entire SparkApplication CR processing reconciliation loop
This can eventually lead to `etcd` errors when we trying to patch / update the CR

**Proposed changes:**
- implement a upper bound limit to the number of tracked executor

## Change Category
Indicate the type of change by marking the applicable boxes:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### Rationale

Submitting apps and tracking driver pods is the highest priority of the spark-operator
1 app going haywire should not hinder the entirety of the cluster

## Checklist
Before submitting your PR, please review the following:

- [X] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [X] I have added tests that prove my changes are effective or that my feature works.
- [X] Existing unit tests pass locally with my changes.

### Additional Notes

This feature has been live on our fork for the past year -> https://github.com/spotinst/spark-on-k8s-operator/pull/8

### CODE REVIEW

**Comment by ImpSy** [comment]
Hey @andreyvelich, @jacobsalway, @mwielgus

Could I please have a review on this PR ?

**Comment by ImpSy** [comment]
@ChenYi015 I've done the change you requested :)
I know that you were not tag on this so thanks for the review 👍

File: cmd/operator/controller/start.go
Comment (vara-bonthu) [inline cmd/operator/controller/start.go:137]: Is this value configurable to users? if yes, how do they do it with this change?

**Comment by ImpSy** [comment]
@vara-bonthu I've added the possibility to customize the value in the chart following your comment
Thanks for the review 🙏

**Review by ChenYi015 — APPROVED** [review]

**Comment by ChenYi015** [comment]
LGTM. Will wait for another approval.

**Comment by ImpSy** [comment]
@ChenYi015 I case it blocked the merge, I've rebase the PR to fix the conflicts

**Comment by ImpSy** [comment]
@ vara-bonthu could you re-review this please 🙏 ?

**Comment by ImpSy** [comment]
@vara-bonthu could you re-review this please 🙏 ?

**Comment by ImpSy** [comment]
@ChenYi015 It's been more than a week that you approve the PR, could we merge it now ?

**Pull request merged**
